### PR TITLE
Address issues from testing with WordPress 5.0

### PR DESCRIPTION
**Pull request by kienstra**
This addresses issues from #1604

### CODE REVIEW

**Comment by kienstra** [comment]
**Request For Review**

Hi @westonruter,
Could you please review this PR, which addresses issues in https://github.com/Automattic/amp-wp/issues/1604#issuecomment-439160374?

Below are the points in that issue:

# Image Alignment
>2. There's a diff in image alignment on page. Amp version has:

https://github.com/Automattic/amp-wp/pull/1627/commits/93a396771cd7e81e2717cdae1ee16546bd6aa92d addresses this. To see the issue, see [this URL](https://paired-ampconfdemo.pantheonsite.io/2018/07/24/wide-image-alignments-with-classic-editor/?amp)

# Gallery Block
>3. Diff in image rendering/resolution. Amp looks like this:

https://github.com/Automattic/amp-wp/pull/1627/commits/c9e0f03b34b423df011e108d5d8c390bcc2630ea addresses this. To see this issue, go to [this URL](https://paired-ampconfdemo.pantheonsite.io/2018/06/05/375/). And to test it locally, create a Gallery block like that. 

# amp-carousel
> 4. These images on the Non-AMP site:
It's expected to have [those images](https://paired-ampconfdemo.pantheonsite.io/wp-admin/post.php?post=375&action=edit) display as an `<amp-carousel>`, as they're in a Gallery block with 'Display as AMP carousel' selected.

# Twenty Nineteen Image
>6. Blue area appearing at https://paired-ampconfdemo.pantheonsite.io/author/corinne-purtill/?amp. AMP looks like this:

https://github.com/Automattic/amp-wp/pull/1627/commits/d7de7d17bd769d55e5adf95e0b6b2723c0d4932b addresses this.

# Follow Us
>7. "Follow Us" widget is empty on non-AMP pages:

There are AMP components, so they're not expected on non-AMP URLs.

# amp-carousel
>8. Media Gallery Widget photos are poor quality in AMP version:
This is a good catch, and something I should have looked at earlier.

Sometimes, there are `<amp-carousel>` elements with very poor image resolution, like in the *Test Media Gallery Widget* [on this URL](https://paired-ampconfdemo.pantheonsite.io/?amp).

The media gallery widget often has thumbnail-size images. And the `<amp-carousel>` usually has a width of `600` and a height of `480`. So it's not really intended to produce a large `<amp-carousel>`, and users might be surprised by how it looks different in AMP.

So https://github.com/Automattic/amp-wp/pull/1627/commits/1afa46b39520b0281b8a9910c6e3d796e7fb893a prevents outputting an `<amp-carousel>` if the `gallery` shortcode (or widget) has a `size` attribute of `thumbnail`.

File: includes/embeds/class-amp-gallery-embed.php [hunk 1]
```diff
@@ -17,6 +17,7 @@ class AMP_Gallery_Embed_Handler extends AMP_Base_Embed_Handler {
 	 */
 	public function register_embed() {
 		add_filter( 'post_gallery', array( $this, 'maybe_override_gallery' ), 10, 2 );
+		add_filter( 'wp_enqueue_scripts', array( $this, 'enqueue_styles' ) );
```
Comment (westonruter) [inline includes/embeds/class-amp-gallery-embed.php]:
```suggestion
		add_action( 'wp_enqueue_scripts', array( $this, 'enqueue_styles' ) );
```

File: includes/embeds/class-amp-gallery-embed.php [hunk 2]
```diff
@@ -224,4 +234,21 @@ public function render( $args ) {
...
+	/**
+	 * Enqueues the Gallery block styling.
+	 *
+	 * It would be better to enqueue this in AMP_Gallery_Block_Sanitizer,
+	 * but by the time that runs, it's too late to call wp_enqueue_style().
+	 *
+	 * @return void
+	 */
+	public function enqueue_styles() {
+		wp_enqueue_style(
```
Comment (westonruter) [inline includes/embeds/class-amp-gallery-embed.php]: Not a big deal, but why have the external CSS file here but not in `add_twentynineteen_image_styles`? I'm fine with having such a small amount of CSS just being inline in the PHP.

Comment (westonruter) [reply includes/embeds/class-amp-gallery-embed.php]: Looks fine. And using an anonymous function works well for me.

Comment (kienstra) [reply includes/embeds/class-amp-gallery-embed.php]: Thanks, how does https://github.com/Automattic/amp-wp/pull/1627/commits/6b7d305ced47864ecf0c898338889e4857da7cc3 look?

**Pull request merged**